h2o-py/h2o/frame.py

```diff
@@ -1011,6 +1011,34 @@ def set_levels(self, levels):
         return H2OFrame._expr(expr=ExprNode("setDomain", self, False, levels), cache=self._ex._cache)
 
 
+    def rename(self, columns=None):
+        """
+        Change names of columns in the frame.
+
+        Dict key is an index or name of the column whose name is to be set.
+        Dict value is the new name of the column.
+
+        :param columns: dict-like transformations to apply to the column names
+        """
+        assert_is_type(columns, None, dict)
+        new_names = self.names
+        ncols = self.ncols
+
+        for col, name in columns.items():
+            col_index = None
+            if is_type(col, int) and (-ncols <= col < ncols):
+                col_index = (col + ncols) % ncols  # handle negative indices
+            elif is_type(col, str) and col in self.names:
+                col_index = self.names.index(col)  # lookup the name
+
+            if col_index is not None:
+                new_names[col_index] = name
+
+        assert_satisfies(new_names, len(new_names) == self.ncol)
```

Comment:
Is there a reason to use this instead of calling `set_names`. I think both of the functions have a place in the API

Reply:
We should not remove it, removing would break the code for the users. I think we can keep `set_name`.

I would suggest to replace 
   assert_satisfies(new_names, len(new_names) == self.ncol)
   self._ex = ExprNode("colnames=", self, range(self.ncol), new_names)  # Update-in-place, but still lazy
with a call of the `set_names` function